PageViewController.swift

```diff
@@ -0,0 +1,211 @@
+        
+        
+        skipButton.setTitle("Skip", for: .normal)
+        skipButton.setTitleColor(.gray, for: .normal)
+        skipButton.frame = CGRect(x: 30, y: 100, width: 50, height: 30)
+        view.addSubview(skipButton)
+        skipButton.addTarget(self, action: #selector(self.skipClicked(_:)), for: .touchUpInside)
+        skipButton.translatesAutoresizingMaskIntoConstraints = false
+        skipButton.centerYAnchor.constraint(equalTo: pageControl.centerYAnchor).isActive = true
+        skipButton.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: 24).isActive = true
+        skipButton.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor).isActive = true
+        
+        let closeButton = UIButton()
+        closeButton.setTitle("X", for: .normal)
+        closeButton.setTitleColor(.gray, for: .normal)
+        closeButton.frame = CGRect(x: 30, y: 100, width: 50, height: 30)
+        view.addSubview(closeButton)
+        closeButton.addTarget(self, action: #selector(self.close(_:)), for: .touchUpInside)
+        closeButton.translatesAutoresizingMaskIntoConstraints = false
+        closeButton.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: 24).isActive = true
+        closeButton.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
+        
+        
+        getStarted.setImage(UIImage(named: "getStarted"), for: .normal)
+        getStarted.frame = CGRect(x: 30, y: 100, width: 186, height: 43)
+        view.addSubview(getStarted)
+        getStarted.addTarget(self, action: #selector(self.close(_:)), for: .touchUpInside)
+        getStarted.translatesAutoresizingMaskIntoConstraints = false
+        getStarted.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+        getStarted.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -130).isActive = true
+        getStarted.isHidden = true
+        getStarted.layer.opacity = 0
+        
+    }
+    
+    @objc func close(_: UIButton) {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+
+    @objc func skipClicked(_: UIButton) {
+        
+        for i in currentIndex...pages.count-1 {
+            setViewControllers([pages[i]], direction: .forward, animated: true, completion: nil)
+        }
+        pageControl.currentPage = pages.count - 1
+        
+    }
+    
+    @objc func nextClicked(_: UIButton){
+        print("Nexting now")
+        moveToNextPage()
+    }
+    
+    func moveToNextPage() {
+        
+        if currentIndex != pages.count - 1 {
+            setViewControllers([pages[currentIndex + 1]], direction: .forward, animated: true, completion: nil)
+        }
+        pageControl.currentPage += 1
+        
+    }
+    
+    func configurePageControl() {
+        let initialPage = 0
+        let page1 = StudyPactOnboardingViewController()
```

Comment:
Since each of the onboardings view are very similar, maybe just create a singular one of these with an init that has parameters like BearImage, Subtitle, etc.